src/world/partner/bow.c

```diff
@@ -97,14 +104,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
+            }
+            npc->pos.x = playerStatus->position.x - D_802BE0E4;
+            npc->pos.y = playerStatus->position.y - D_802BE0E8;
+            npc->pos.z = playerStatus->position.z - D_802BE0EC;
+
+            distance = dist2D(0.0f, 0.0f, partnerActionStatus->stickX, partnerActionStatus->stickY);
+            if ((collisionStatus->currentFloor < 0) || distance > 10.0f ||
+                 partnerActionStatus->pressedButtons & (BUTTON_B | BUTTON_C_DOWN) || playerStatus->flags & 0x800) {
+                if (func_802BD540_323E90() < 0) {
+                    script->functionTemp[0]++;
+                    script->functionTemp[1] = 3;
+                    script->functionTemp[2] = playerStatus->statusMenuCounterinputEnabledCounter;
+                }
+            }
+            break;
+        case 3:
+            if (script->functionTemp[1] == 0) {
+                if (script->functionTemp[2] < playerStatus->statusMenuCounterinputEnabledCounter) {
+                    script->functionTemp[0] = 2;
+                    break;
+                }
+            } else {
+                script->functionTemp[1]--;
+                break;
+            }
+
+            sfx_play_sound_at_npc(SOUND_BOW_APPEAR, 0, -4);
+            func_802BDDF0_324740(npc);
+            return ApiStatus_DONE1;
+    }
+    return ApiStatus_BLOCK;
+}
+
+void func_802BDDF0_324740(Npc* partner) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    s32 actionState;
+
+    playerStatus->alpha1 = 255;
+    func_8003D624(partner, 0, 0, 0, 0, 0, 0);
+    partner->renderMode = 0x11;
+    get_shadow_by_index(partner->shadowIndex)->unk_05 = playerStatus->alpha1 >> 1;
+
+    if (D_802BE0C4 != 0) {
+        enable_player_input();
+    }
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BDDF0_324740);
+    playerStatus->flags &= ~0x8002;
+    partner->flags &= ~0x42;
+    D_802BE0C4 = 0;
+    actionState = ACTION_STATE_IDLE;
+    if (playerStatus->flags & 0x800) {
+        actionState = ACTION_STATE_HIT_LAVA;
+    }
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BDF08_324858);
+    set_action_state(actionState);
+    partnerActionStatus->actionState.b[0] = 0;
+    partnerActionStatus->actionState.b[3] = 0;
+    playerStatus->flags &= ~0x100;
+    partner_clear_player_tracking(partner);
+    D_802BE0C0 = 0;
+}
+
+ApiStatus func_802BDF08_324858(Evt* script, s32 isInitialCall) {
+    Npc* partner = script->owner2.npc;
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BDF64_3248B4);
+    if (isInitialCall) {
+        partner_init_put_away(partner);
+        if (D_802BE0C0 != 0) {
+            sfx_play_sound_at_npc(SOUND_BOW_APPEAR, 0, -4);
+        }
+        func_802BDDF0_324740(partner);
+    }
+    return partner_put_away(partner) != 0;
+}
+
+void func_802BDF64_3248B4(Npc* partner) {
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+
+    if (D_802BE0C0 != 0) {
```

Comment:
This seems to be a bool?
```suggestion
    if (D_802BE0C0) {
```